### PR TITLE
Calibrate with fixed r

### PR DESCRIPTION
**Pull request by LouisChrist**
[edit by @elithrar]: Addresses https://github.com/elithrar/simple-scrypt/issues/12

### CODE REVIEW

**Comment by elithrar** [comment]
@FiloSottile - mind taking a look at this as well? More eyes always good.

**Comment by elithrar** [comment]
Failed in Go 1.7 - restarted the build. 

```sh
The command "go tool vet ." exited with 0.
21.19s$ go test -v ./...
=== RUN   TestGenerateRandomBytes
--- PASS: TestGenerateRandomBytes (0.00s)
=== RUN   TestGenerateFromPassword
--- PASS: TestGenerateFromPassword (9.47s)
=== RUN   TestCompareHashAndPassword
--- PASS: TestCompareHashAndPassword (0.18s)
=== RUN   TestCost
--- PASS: TestCost (0.06s)
=== RUN   TestDecodeHash
--- PASS: TestDecodeHash (0.00s)
=== RUN   TestCalibrate
--- FAIL: TestCalibrate (10.40s)
	scrypt_test.go:140: GenerateFromPassword with scrypt.Params{N:131072, R:8, P:1, SaltLen:16, DKLen:32} took 472.320073ms (<nil>)
	scrypt_test.go:140: GenerateFromPassword with scrypt.Params{N:131072, R:8, P:1, SaltLen:16, DKLen:32} took 506.033718ms (<nil>)
	scrypt_test.go:140: GenerateFromPassword with scrypt.Params{N:65536, R:8, P:1, SaltLen:16, DKLen:32} took 230.961435ms (<nil>)
	scrypt_test.go:145: 2. GenerateFromPassword was too fast (wanted around 500ms, got 230.961435ms) with scrypt.Params{N:65536, R:8, P:1, SaltLen:16, DKLen:32}.
	scrypt_test.go:140: GenerateFromPassword with scrypt.Params{N:32768, R:8, P:4, SaltLen:16, DKLen:32} took 657.942582ms (<nil>)
	scrypt_test.go:140: GenerateFromPassword with scrypt.Params{N:16384, R:8, P:8, SaltLen:16, DKLen:32} took 495.989112ms (<nil>)
	scrypt_test.go:140: GenerateFromPassword with scrypt.Params{N:8192, R:8, P:17, SaltLen:16, DKLen:32} took 463.957228ms (<nil>)
	scrypt_test.go:140: GenerateFromPassword with scrypt.Params{N:4096, R:8, P:34, SaltLen:16, DKLen:32} took 520.679261ms (<nil>)
	scrypt_test.go:140: GenerateFromPassword with scrypt.Params{N:512, R:8, P:280, SaltLen:16, DKLen:32} took 527.884269ms (<nil>)
FAIL
exit status 1
FAIL	github.com/elithrar/simple-scrypt	20.183s
The command "go test -v ./..." exited with 1.
```

**Comment by LouisChrist** [comment]
It works perfectly fine on windows, but on linux not at all. The measured runtime of GenerateFromPassword() with the same parameters varies on my linux system by about 50ms to 100ms.

**Comment by elithrar** [comment]
The problem is that in a CI, or other contended environment, a +/- 250ms difference on a 500ms target isn't that absurd given the "expense" of scrypt.

Calibration, being what it is, isn't perfect. Proposing:

- The 500ms tests should be more forgiving (+/- 75%)
- Add additional tests (e.g. 1500ms) to help with accuracy at higher absolute run times.

**Comment by LouisChrist** [comment]
This should work better. It's slower but the results are more consistent. I ran the tests 20 times and had no failures. 

This might also fix #8. TestCalibrate fails around 80% percent of the time with the old Calibrate function one my computer.

File: scrypt_test.go [hunk 1]
```diff
@@ -140,7 +143,7 @@ func TestCalibrate(t *testing.T) {
 		}
 		if dur < timeout/2 {
 			t.Errorf("%d. GenerateFromPassword was too fast (wanted around %s, got %s) with %#v.", testNum, timeout, dur, p)
-		} else if timeout*2 < dur {
+		} else if timeout+timeout/2 < dur {
 			t.Errorf("%d. GenerateFromPassword took too long (wanted around %s, got %s) with %#v.", testNum, timeout, dur, p)
```
Comment (elithrar) [inline scrypt_test.go]: Looks good, but one last ask: can you change the error message to detail what "around" means - e.g. "expected between {lower bound} and {upper bound}".

Comment (LouisChrist) [reply scrypt_test.go]: Yes of course

**Pull request merged**